disk/disk.go

```diff
@@ -18,130 +17,160 @@ type Disk struct {
 	Root string
 }
 
+type BlockReaderStream struct {
+	blockIndex int
+	blockOffset int
+	file io.ReadSeeker
+}
+
+func (brs *BlockReaderStream) NextBlock() (*Block, error) {
+	block := newBlock()
+	err := readBlock(brs.file, block, brs.blockIndex)
+	block.left = brs.blockOffset
```

Comment:
block.StartFrom()? sometimes we change left field directly, sometimes it is changed in member function. I think it may not good to change left field directly.